### DISCUSSION TITLE
Forwarding HLS request to AWS S3 and Resolve the Playback problem

### DISCUSSION THREAD

**Discussion by mekya** [ant-media/Ant-Media-Server#6078] [General]
Hi @yashtandon113,

I want to share our discussion here because it may help some friends.

You were testing to upload the HLS chunks to S3 directly while live stream is active. Additionally, user would like to play the stream through the embedded player we provide. 

I've investigate this case and it turns out that the problem is about CORS and the server address should be entered in S3 bucket configuration. 

Let me tell how I resolved the issue step by step

1. I configure the m3u8 extension and set the S3 address in the Advanced Settings of the Application as in the image. (I've used the @Mohit-3196 test bucket 😀 ) Pay attention that there is no trailer slash in the S3 bucket url because AMS adds a trailer slash and AWS returns 403 if there are two slashes.
 
<img width="800" alt="Screenshot 2024-02-01 at 10 43 45" src="https://github.com/ant-media/Ant-Media-Server/assets/3456251/133b795f-724d-4fbb-85e0-f2ef625ad3dd">

2. I've visited  AWS S3 > {Bucket Name} > Permissions tab and scroll bottom to the CORS policy. 

<img width="800" alt="Screenshot 2024-02-01 at 10 53 24" src="https://github.com/ant-media/Ant-Media-Server/assets/3456251/545d0c0b-9ebe-46c7-8d26-86a6599d10f4">

3.  I've entered the Ant Media Server address where I want to play the stream. Wildcard('*') is not working there, we need to set the `AllowedOrigin` specifically because of CORS policy. Check the image below

<img width="800" alt="Screenshot 2024-02-01 at 10 48 44" src="https://github.com/ant-media/Ant-Media-Server/assets/3456251/ad95e9cd-7578-4ca7-be3a-0b74409f97b1">

4. Then try of play the stream directly with embedded player on the application 🎉  
<img width="800" alt="Screenshot 2024-02-01 at 10 51 02" src="https://github.com/ant-media/Ant-Media-Server/assets/3456251/a8ac4f1c-c056-4f78-a002-be491c8b7856">

It's working for me. Please check that if it's working for you. 

Let me know if I can help for anything. 

Here is the documentation about [S3 Integration and HTTP Forwarding](https://antmedia.io/docs/guides/recording-live-streams/s3-integration-http-forwarding/)
 
Cheers
Oguz

**Comment by yashtandon113** [comment]
@mekya 
Thanks for checking this out.

It is working as expected now after putting the server address in the CORS policy.
